### PR TITLE
Merge part of feature/basiccrud for entites Class, Pupil

### PR DESCRIPTION
**Pull request by dima-melnyk**

### CODE REVIEW

File: GradeBook.API/Controllers/ClassController.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+using GradeBook.DataAccess.Entities;
+using AutoMapper;
+
+namespace GradeBook.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ClassController : ControllerBase
+    {
+        private IClassService _classService;
```
[the remark this answers is not in the document]
Comment (dima-melnyk) [reply GradeBook.API/Controllers/ClassController.cs:15]: Sure

File: GradeBook.API/Controllers/ClassController.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        private readonly IMapper _mapper;
+
+        public ClassController(IClassService classService, IMapper mapper)
+        {
+            _classService = classService;
+            _mapper = mapper;
+        }
+
+        [HttpGet("{id}")]
+        public async Task<ClassToView> GetClass([FromRoute] int id)
```
Comment (Kolezhniuk) [inline GradeBook.API/Controllers/ClassController.cs:25]: remove async await

File: GradeBook.BusinessLogic/Models/ClassToView.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿namespace GradeBook.BusinessLogic.Models
+{
```
Comment (Kolezhniuk) [inline GradeBook.BusinessLogic/Models/ClassToView.cs:2]: Create folder Read/Write

File: GradeBook.BusinessLogic/Services/ClassService.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+        public async Task CreateClass(Class newClass)
+        {
+            await _repository.AddAsync(newClass);
+        }
+
+        public async Task DeleteClass(int id)
+        {
+            var model = await _repository.GetByIdAsync(id);
+            await _repository.RemoveAsync(model);
```
Comment (Kolezhniuk) [inline GradeBook.BusinessLogic/Services/ClassService.cs:29]: investigate if it possible to remove entity just by id

File: GradeBook.BusinessLogic/Services/GradeService.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            await _repository.RemoveAsync(model);
+        }
+
+        public async Task<GradeToView> GetGrade(int id)
+        {
+            var model = await _repository.GetByIdAsync(id);
+            return _mapper.Map<GradeToView>(model);
+        }
+
+        public List<GradeToView> GetGrades(GradeQuery query)
```
Comment (Kolezhniuk) [inline GradeBook.BusinessLogic/Services/GradeService.cs:48]: make it async

Comment (Kolezhniuk) [reply GradeBook.BusinessLogic/Services/GradeService.cs:48]: =>

File: GradeBook.BusinessLogic/Services/LessonService.cs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+        public async Task CreateLesson(Lesson newLesson)
+        {
+            await _repository.AddAsync(newLesson);
+        }
+
+        public async Task DeleteLesson(int id)
+        {
+            var model = await _repository.GetByIdAsync(id);
+            model.Grades.Select(async grade => await _gradeRepository.RemoveAsync(grade));
```
Comment (Kolezhniuk) [inline GradeBook.BusinessLogic/Services/LessonService.cs:34]: batch removing

File: GradeBook.BusinessLogic/Services/TeacherService.cs [hunk 7]
```diff
@@ -0,0 +1,39 @@
+﻿using AutoMapper;
+using GradeBook.BusinessLogic.Interfaces;
+using GradeBook.BusinessLogic.Models;
+using GradeBook.DataAccess.Entities;
+using GradeBook.Repository.Interfaces;
+using System.Threading.Tasks;
+
+namespace GradeBook.BusinessLogic.Services
+{
+    public class TeacherService : ITeacherService
```
Comment (Kolezhniuk) [inline GradeBook.BusinessLogic/Services/TeacherService.cs:10]: TeacherManagement

**Pull request merged**